copyshop.py

```diff
@@ -103,34 +158,97 @@ def translate_commands(commands):
+            if command["data"]["using"] == "accessibility id":
+                c = "el = driver.findElementByAccessibilityId(\"{}\");".format(command["data"]["value"])
+        elif command["method"] == "POST" and command["path"].endswith("/click"):
+            c = "el.click();"
+        elif command["method"] == "POST" and command["path"].endswith("/context"):
+            c = "driver.context(\"{}\");".format(command["data"]["name"])
+        else:
+            c += ": {} {}\n".format(command["method"], command["path"])
+            c += "// " + repr(command["data"])
+
+        java_commands.append(c)
+    return capabilities, java_commands
+
+
+def parse_appium_log_line(line):
+    info = {}
+    pattern = "(\[HTTP\]) --> (GET|POST|PUT|DELETE) (.+)"
+    results = re.search(pattern, line)
+
+    if not results:
+        raise Exception("Command not recognised: {}".format(line))
+
+    info["method"] = results.group(2)
+    info["path"] = unescape(results.group(3))
+    return info
+
 
 def escape(s):
-    s = s.replace("\"", "\\\"")
-    # s = s.replace("'", "\'")
-    return s
+    return s.replace("\"", "\\\"")
 
+def unescape(s):
+    return s.replace("\/", "/")
 
 def main(arguments=None):
 
     parser = argparse.ArgumentParser()
     parser.add_argument("job_urls", nargs="+", help="URL of test to copy.")
 
     args = parser.parse_args(arguments)
-
     for job_url in args.job_urls:
+        parsed = urlparse(job_url)
+        if parsed.netloc == "app.saucelabs.com" or parsed.netloc == "app.eu-central-1.saucelabs.com":
+            _vdc_main(args, job_url)
+        elif parsed.netloc == "app.testobject.com":
+            _rdc_main(args, job_url)
+        else:
+            raise Exception("URL not recognized: {}".format(job_url))
+
+def _vdc_main(args, job_url):
+    url_info = extract_vdc_url_info(job_url)
+
+    info = retrieve_job_info(url_info["domain"], url_info["job_id"])
+
+    log_filename = get_log_filename(url_info["domain"], url_info["job_id"])
+    commands = extract_commands(url_info["domain"], url_info["job_id"],
+                                log_filename)
+    java_commands = translate_commands(commands)
 
-        url_info = extract_url_info(job_url)
+    template = Template(open("./template.java").read())
+    print(job_info_to_java(info, url_info["domain"], java_commands,
+                            template))
 
-        info = retrieve_job_info(url_info["domain"], url_info["job_id"])
+def _rdc_main(args, job_url):
+    url_info = extract_rdc_url_info(job_url)
+    info = retrieve_rdc_appium_logs(url_info["domain"], url_info["username"], url_info["project"], url_info["job_id"])
 
-        log_filename = get_log_filename(url_info["domain"], url_info["job_id"])
-        commands = extract_commands(url_info["domain"], url_info["job_id"],
-                                    log_filename)
-        java_commands = translate_commands(commands)
+    capabilities, java_commands = translate_rdc_commands(info)
```

Comment:
Do we take into account if there is no commands and fail gracefully? We might make the mistake of trying to get a link that has no Appium log and would just crash without saying why.

Reply:
I will add a verification to be sure. However we should probably have a different PR going through all the possible failures and give meaningful errors.

Reply:
Cool yeah. Created https://github.com/rgauthier-sauce/copy-shop/issues/5 to track this.

Reply:
I won't add a verification in this PR because it would require a bit more work, especially I need a link to a report without Appium logs and see what happens. Let's do that in #5.